pkg/apis/cloudingress/v1alpha1/publishingstrategy_types.go

```diff
@@ -0,0 +1,81 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// PublishingStrategySpec defines the desired state of PublishingStrategy
+type PublishingStrategySpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+
+	// DefaultAPIServerIngress defines whether API is internal or external
+	DefaultAPIServerIngress DefaultAPIServerIngress `json:"defaultAPIServerIngress"`
+	//ApplicationIngresws defines whether application ingress is internal or external
+	ApplicationIngress []ApplicationIngress `json:"applicationIngress"`
+}
+
+// DefaultAPIServerIngress defines API ingress
+type DefaultAPIServerIngress struct {
+	// Listening defines internal or external ingress
+	Listening Listening `json:"listening,omitempty"`
+}
+
+// ApplicationIngress defines application ingress
+type ApplicationIngress struct {
+	// Listening defines application ingress as internal or external
+	Listening Listening `json:"listening,omitempty"`
+	// Default defines default value of ingress when cluster installs
+	Default       bool                   `json:"default"`
+	DNSName       string                 `json:"dnsName"`
+	Certificate   corev1.SecretReference `json:"certificate"`
+	RouteSelector metav1.LabelSelector   `json:"routeSelector"`
+}
+
+// Listening defines internal or external api and ingress
+type Listening string
```

Comment:
Is this type used outside of the const and the two struct instances definitons? I'm questioning how useful this type alias is

Reply:
could just be used as const strings